### ISSUE TITLE
number of maxpooling and upsampling in VGGSegnet and VGGUnet is not same

### ISSUE THREAD

**Issue by gangooteli** [divamgupta/image-segmentation-keras#18]
The number of times maxpooling used is not equal to number of times upsampling in VGGsegnet and VGGUnet due to which output image dimensions are not equal to input image dimensions.
In this case we need to use resize as you have done in predict.
If # of maxpool == # of upsampling layer, then input image dim == output image dim and no need of resize.

using resize might lead to less accuracy

**Comment by divamgupta** [comment]
Having more upsampling layers would increase the number of parameters which are not initialized by any pretrained model. That would make it much harder to train.

**Comment by gangooteli** [comment]
If we use vggnet segnet pre-trained weights, it will be applied to VGG16 layers i.e. Decoder part(or Downsamling part). The upsampling part is learned from scratch. In that in implementation of VGGSegnet, we can add one more upsampling layer to get the image dim == output dim.
And we need note to use resize.
What is your thoughts on that ?

**Comment by divamgupta** [comment]
A better idea would be to use less encoder layers from VGGNet. You could also increase the upsampling layers with skip connections for easy training.
